### PR TITLE
[RW-9741][risk=low] AoU Docker image upgrade: Hail, download extension, alias Cromshell

### PR DESCRIPTION
**Pull request by yonghaoy**

### CODE REVIEW

File: terra-jupyter-aou/Dockerfile [hunk 1]
```diff
@@ -47,7 +47,7 @@ RUN jupyter nbextension enable --py igv --sys-prefix
 ENV PYTHONPATH $PYTHONPATH:/usr/lib/spark/python
 
 ENV PYSPARK_PYTHON=python3
-ENV HAIL_VERSION=0.2.109
+ENV HAIL_VERSION=0.2.112
```
Comment (tarekmamdouh) [inline terra-jupyter-aou/Dockerfile:50]: Dan is already updating this in his PR https://github.com/DataBiosphere/terra-docker/pull/408

File: terra-jupyter-aou/Dockerfile [hunk 2]
```diff
@@ -266,6 +266,16 @@ RUN PERL_MM_USE_DEFAULT=1 cpan Text::Soundex
...
 
+
+# Install Cromshell
+RUN cromshell-alpha version
+
+# Remove workspace_cromwell script as it may allow user bypass AoU to create Cromwell via Leo directly
+rm -f /usr/local/bin/workspace_cromwell.py
+
+# Downgrade jupyter_contrib_nbextensions as the newer version breaks snippet_menu extension
+pip install --force-reinstall -v "jupyter_contrib_nbextensions==0.5.1"
```
Comment (tarekmamdouh) [inline terra-jupyter-aou/Dockerfile:277]: RUN pip install

File: terra-jupyter-aou/Dockerfile (hunk 1 above)
Comment (yonghaoy) [reply terra-jupyter-aou/Dockerfile:50]: I left a comment in that PR and let Dan know this PR also includes Hail upgrade

**Comment by yonghaoy** [comment]
This is currently blocked by https://github.com/DataBiosphere/terra-docker/pull/394#issuecomment-1472621585

**Comment by tarekmamdouh** [comment]
bolt lmm is also broken again, it should be available again next week
https://community.dnanexus.com/s/question/0D5t000004SCMJxCAP/boltlmm-download-link-broken

File: terra-jupyter-aou/Dockerfile [hunk 3]
```diff
@@ -266,6 +266,16 @@ RUN PERL_MM_USE_DEFAULT=1 cpan Text::Soundex
 RUN curl -L http://www.repeatmasker.org/RepeatMasker/RepeatMasker-4.1.4.tar.gz \
   | (cd /usr/local; tar -xz)
 
+
+# Install Cromshell
+RUN cromshell-alpha version
```
Comment (tarekmamdouh) [inline terra-jupyter-aou/Dockerfile:271]: this doesn't work, I got this error locally

Step 44/49 : RUN cromshell-alpha version
 ---> Running in d3cf1b1fa50d
/bin/sh: 1: cromshell-alpha: not found
The command '/bin/sh -c cromshell-alpha version' returned a non-zero code: 127

Comment (yonghaoy) [reply terra-jupyter-aou/Dockerfile:271]: right, we decide to use Cromshell-beta: https://pmi-engteam.slack.com/archives/CHRN2R51N/p1678994326910319
I will revert this change

File: terra-jupyter-base/Dockerfile [hunk 4]
```diff
@@ -112,12 +112,6 @@ RUN pip3 -V \
     # For gcloud alpha storage support.
     && pip3 install google-crc32c --target /usr/lib/google-cloud-sdk/lib/third_party
 
-# install Cromshell 2.0
```
Comment (tarekmamdouh) [inline terra-jupyter-base/Dockerfile:115]: @yonghaoy I think we need to bring these lines back

**Comment by tarekmamdouh** [comment]
The changes LGTM, hoping the build works this time

**Review by tarekmamdouh — APPROVED** [review]

**Pull request merged**